Sources/NIOPosix/System.swift

```diff
@@ -644,12 +663,121 @@ internal enum Posix {
     }
 
     @inline(never)
-    internal static func unlink(pathname: String) throws {
+    internal static func lstat(pathname: String, outStat: UnsafeMutablePointer<stat>) throws {
+        _ = try syscall(blocking: false) {
+            sysLstat(pathname, outStat)
+        }
+    }
+
+    @inline(never)
+    public static func symlink(pathname: String, destination: String) throws {
+        _ = try syscall(blocking: true) {
+            sysSymlink(destination, pathname)
+        }
+    }
+
+    @inline(never)
+    public static func readlink(pathname: String, outPath: UnsafeMutablePointer<CChar>, outPathSize: Int) throws {
+        _ = try syscall(blocking: true) {
+            sysReadlink(pathname, outPath, outPathSize)
+        }
+    }
+
+    @inline(never)
+    public static func unlink(pathname: String) throws {
         _ = try syscall(blocking: false) {
             sysUnlink(pathname)
         }
     }
 
+    @inline(never)
+    public static func mkdir(pathname: String, mode: mode_t) throws {
+        _ = try syscall(blocking: true) {
```

Comment:
Blocking should be `false`, this cannot return `EAGAIN` or `EWOULDBLOCK`.

Reply:
removed for now